src/Sulu/Component/HttpCache/Handler/TagsHandler.php

```diff
@@ -48,51 +43,30 @@ class TagsHandler implements
 
     /**
      * @param ProxyClientInterface $proxyClient
-     * @param ContentTypeManagerInterface $contentTypeManager
+     * @param ReferenceStorePoolInterface $referenceStore
      */
-    public function __construct(
-        ProxyClientInterface $proxyClient,
-        ContentTypeManagerInterface $contentTypeManager
-    ) {
+    public function __construct(ProxyClientInterface $proxyClient, ReferenceStorePoolInterface $referenceStore)
+    {
         $this->proxyClient = $proxyClient;
-        $this->contentTypeManager = $contentTypeManager;
+        $this->referenceStore = $referenceStore;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function invalidateStructure(StructureInterface $structure)
     {
         $this->structuresToInvalidate[$structure->getUuid()] = $structure;
     }
 
+    /**
+     * {@inheritdoc}
+     */
     public function updateResponse(Response $response, StructureInterface $structure)
     {
-        $tags = [
-            $this->getBanKey($structure->getUuid()),
-        ];
-
-        foreach ($structure->getProperties(true) as $property) {
-            foreach ($this->getReferencedUuids($property) as $uuid) {
-                $tags[] = $this->getBanKey($uuid);
-            }
-        }
+        $tags = array_merge([$structure->getUuid()], $this->referenceStore->getReferences());
```

Comment:
But the other fields from the `ReferenceStore` will always have a type, or am I wrong? If I read the code correctly it will be `content-uuid` for all other pages, except for this single one here.